app/models/stocks/price.rb

```diff
@@ -0,0 +1,35 @@
+require "json"
+require "redis"
+
+class Stocks::Price
+  REDIS_HOST = "redis://redis:6379/"
+
+  attr_accessor :open, :high, :low, :close, :volume,
+    :adj_high, :adj_low, :adj_close, :adj_open, :adj_volume,
+    :split_factor, :dividend, :symbol, :exchange, :date
+
+  def assign_attrs(attrs = {})
+    attrs.each_pair do |k, v|
+      instance_variable_set("@#{k}", v) if v.present? && respond_to?(k)
+    end
+  end
+
+  def initialize(attrs = {})
+    assign_attrs(attrs) unless attrs.blank?
+  end
+
+  def get_end_of_day_price(symbol:, date: "latest")
+    # date could be "latest" or "YYYY-MM-DD"
+    redis = Redis.new(url: REDIS_HOST)
+    cached = redis.get("eod_#{symbol.downcase}_#{date}")
+    assign_attrs(JSON.parse(cached)) if cached.present?
```

Comment:
Please use the Rails cache wrappers rather than directly connecting to a running redis instance.

see the guides here:  https://guides.rubyonrails.org/caching_with_rails.html#activesupport-cache-rediscachestore